IndicoToolkit/fileprocessing.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace IndicoToolkit
+{
+    public class FileProcessing
+    {
+        public List<string> FilePaths; 
+        public List<string> DefaultAcceptedTypes; 
+        
+        public FileProcessing ( List<string> filePathsInput = null)
+        {
+            FilePaths = filePathsInput ?? new List<string>();
+            DefaultAcceptedTypes = new List<string>() {"*.pdf", "*.doc", "*.docx", "*.tiff", "*.tif", "*.png"};
+        }
+
+        public void GetFilePathsFromDir(string pathToDir,
+                                        IEnumerable<string> acceptedTypes = null,
+                                        bool topDirectoryOnly = true)
+        {
+            acceptedTypes = acceptedTypes ?? DefaultAcceptedTypes; 
+            SearchOption searchOption = SearchOption.TopDirectoryOnly;
+            if (!topDirectoryOnly)
+            {
+                searchOption = SearchOption.AllDirectories;
+            }
+
+            FilePaths = FileSearch( pathToDir, acceptedTypes.Distinct().ToList(), searchOption);
+            
+            // Print statment about files added or not added
+            int dirSize = Directory.GetFiles(pathToDir, "*", searchOption).Length;
+            Console.WriteLine($"Found {FilePaths.Count} valid files and {dirSize - FilePaths.Count} paths with invalid suffixes. Valid suffix:");
+        }
+
+        private List<string> FileSearch(string pathToDir,
+                                        IEnumerable<string> acceptedTypes,
+                                        SearchOption searchOption = SearchOption.TopDirectoryOnly)
+        {
+            List<string> result = new List<string>();
+            foreach (var acceptedType in acceptedTypes)
+            {
+                result.AddRange(Directory.GetFiles(pathToDir, acceptedType, searchOption));
+            }
+
+            return result;
+        }
+
+        public IEnumerable<List<string>> BatchFiles(int batchSize = 10) 
+        {
+            for (int i = 0; i < FilePaths.Count; i = i + batchSize)
+            {
+                int end = Math.Min(batchSize, FilePaths.Count - i);
+                yield return FilePaths.GetRange(i, end);
+            }
```

Comment:
could this just be (i, i + batchSize)? The Math.Min logic doesn't look right to me and I don't think you test it. i.e. if my batch size is 10 and I'm on the last loop of a 100 docs,  min(10, 100 - 90) when my end should be 100 right?